pandas/_libs/tslibs/c_timestamp.pyx

```diff
@@ -253,6 +253,13 @@ cdef class _Timestamp(datetime):
         elif is_array(other):
             if other.dtype.kind in ['i', 'u']:
                 raise integer_op_not_supported(self)
+            if other.dtype.kind == "m":
+                if self.tz is None:
```

Comment:
why do we return ndarray's here? and not DTI / TDI?

Reply:
decided against DTA/TDA for dependency-graph